code/modules/modular_computers/file_system/reports/report.dm

```diff
@@ -20,40 +22,24 @@
 	. = ..()
 
 /*
-Access stuff. The report's access/access_edit should control whether it can be opened/submitted.
-For field editing or viewing, use the field's access/access_edit permission instead.
-The access system is based on "access patterns", lists of access values. 
-A user needs all access values in a pattern to be granted access.
-A user needs to only match one of the potentially several stored access patterns to be granted access.
-You must have access to have edit access.
-
-This proc resets the access to the report, resulting in just one access pattern for access/edit.
+This proc resets the access to the report, resulting in just one access pattern for read/write.
 Arguments can be access values (numbers) or lists of access values.
 If null is passed to one of the arguments, that access type is left alone. Pass list() to reset to no access needed instead.
 The recursive option resets access to all fields in the report as well.
 If the override option is set to 0, the access supplied will instead be added as another access pattern, rather than resetting the access.
 */
-/datum/computer_file/report/proc/set_access(access, access_edit, recursive = 1, override = 1)
-	if(access)
-		if(!islist(access))
-			access = list(access)
-		override ? (src.access = list(access)) : (src.access += list(access))  //Note that this is a list of lists.
-	if(access_edit)
-		if(!islist(access_edit))
-			access_edit = list(access_edit)
-		override ? (src.access_edit = list(access_edit)) : (src.access_edit += list(access_edit))
+/datum/computer_file/report/proc/set_access(read_access, write_access, recursive = 1, override = 1)
+	if(read_access)
+		if(!islist(read_access))
+			read_access = list(read_access)
+		override ? (src.read_access = list(read_access)) : (src.read_access += list(read_access))  //Note that this is a list of lists.
+	if(write_access)
+		if(!islist(write_access))
+			write_access = list(write_access)
+		override ? (src.write_access = list(write_access)) : (src.write_access += list(write_access))
 	if(recursive)
 		for(var/datum/report_field/field in fields)
-			field.set_access(access, access_edit, override)
-
-//Strongly recommended to use these procs to check for access. They can take access values (numbers) or lists of values.
-/datum/computer_file/report/proc/verify_access(given_access)
-	return has_access_pattern(access, given_access)
-
-/datum/computer_file/report/proc/verify_access_edit(given_access)
-	if(!verify_access(given_access))
-		return //Need access for access_edit
-	return has_access_pattern(access_edit, given_access)
+			field.set_access(read_access, write_access, override, TRUE)
```

Comment:
TBH should rewrite this to use the standard list(list())-accepting notation everywhere else. It's been on my todo for ages now.

Reply:
Done